File: python/lang/otterop/lang/result.py

class Result:
        
    def __init__(self, res, err):
        self._res = res
        self._err = err

    def err(self):
        return self._err

    def unwrap(self):
        return self._res

    @staticmethod
    def of(res, err):
        return Result(res, err)

File: ts/lang/otterop/lang/Result.ts

export class Result<RES, ERR> {

    #res : RES;

    #err : ERR;

    private constructor(res : RES, err : ERR) {
        this.#res = res;
        this.#err = err;
    }

    public err() : ERR {
        return this.#err;
    }

    public unwrap() : RES {
        return this.#res;
    }

    public static of<RES0, ERR0>(res : RES0, err : ERR0) : Result<RES0, ERR0> {
        return new Result<RES0,ERR0>(res, err);
    }
}

